src/components/alarm/AlarmController.h

```diff
@@ -0,0 +1,68 @@
+/*  Copyright (C) 2021 mruss77, Florian
+
+    This file is part of InfiniTime.
+
+    InfiniTime is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    InfiniTime is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <cstdint>
+#include "app_timer.h"
+#include "components/datetime/DateTimeController.h"
+
+namespace Pinetime {
+  namespace System {
+    class SystemTask;
+  }
+  namespace Controllers {
+    class AlarmController {
+    public:
+      AlarmController(Controllers::DateTime& dateTimeController);
+
+      void Init(System::SystemTask* systemTask);
+      void SetAlarm(uint8_t alarmHr, uint8_t alarmMin);
+      void DisableAlarm();
+      void SetOffAlarmNow();
+      uint32_t SecondsToAlarm();
+      void StopAlerting();
+      enum class AlarmState { Not_Set, Set, Alerting };
+      enum class RecurType { None, Daily, Weekdays };
+      uint8_t Hours() const {
+        return hours;
+      }
+      uint8_t Minutes() const {
+        return minutes;
+      }
+      AlarmState State() const {
+        return state;
+      }
+      RecurType Recurrence() const {
+        return recurrence;
+      }
+      void SetRecurrence(RecurType recurType) {
+        recurrence = recurType;
+      }
+
+    private:
+      Controllers::DateTime& dateTimeController;
+      System::SystemTask* systemTask = nullptr;
+      uint8_t hours;
+      uint8_t minutes;
```

Comment:
These are uninitialized, and should probably be set to around 7 o'clock by default instead of midnight.

Reply:
Agree, initialized hours to 7 and minutes to 0